lib/linter.js

```diff
@@ -243,9 +243,37 @@ function getDirectiveComments(filename, ast, ruleMapper) {
                     }
                     break;
 
-                case "eslint-disable":
-                    disableDirectives.push(...createDisableDirectives("disable", comment.loc.start, directiveValue));
+                case "eslint-disable": {
+                    const directiveRules = createDisableDirectives("disable", comment.loc.start, directiveValue);
+
+                    // validate disable directives
+                    directiveRules.forEach(directive => {
+                        const ruleId = directive.ruleId;
+
+                        if (ruleId !== null) {
+                            try {
+                                validator.validateRuleOptions(ruleMapper(ruleId), ruleId, [0]);
```

Comment:
I think it should check only existence.

For that, the fact that `ruleMapper` returns a stub rule if the rule didn't exist is an obstacle. There are some ways:

- add a new parameter like `ruleExists(ruleId)` to check existence.
- separate [createMissingRule](https://github.com/eslint/eslint/blob/5dad0b1d80c9cf380c49f46266c35d461d3cecad/lib/rules.js#L28) from the `Rules` class then use it on only [lib/linter.js#L707](https://github.com/eslint/eslint/blob/5dad0b1d80c9cf380c49f46266c35d461d3cecad/lib/linter.js#L707).

Reply:
surely it is. I didn't even noticed this behavior.

Reply:
what do you think if not returning a stub rule?

Reply:
Sounds good!